test/integration_test/basic_test.go

```diff
@@ -440,6 +465,37 @@ func BasicInstallWithNodeAffinity(tc *types.TestCase) func(*testing.T) {
 	}
 }
 
+// 1. Deploy PX and verify the token for px stored in the k8s secret correctly mounted inside px runc container.
+// 2. The cluster spec set the token expiration time to be 3 min. So the token is supposed to refresh after 1.5min.
+// 3. Wait for 1.5min. Verify the token is refreshed and corrected mounted inside px run container with retry.
+func BasicInstallWithPxSaTokenRefresh(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
+
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+		pxSaSecret, err := coreops.Instance().GetSecret(pxutil.PortworxServiceAccountTokenSecretName, cluster.Namespace)
+		require.NoError(t, err)
+		expectedToken := pxSaSecret.Data[core.ServiceAccountTokenKey]
+		cmd := "runc exec portworx cat /var/run/secrets/kubernetes.io/serviceaccount/token"
+		actualToken, stderr, err := ci_utils.RunPxCmd(cmd)
+		require.Empty(t, stderr)
+		require.Nil(t, err)
+		require.Equal(t, actualToken, expectedToken)
+
+		time.Sleep(time.Duration(90) * time.Second)
+
+		pxSaSecret, err = coreops.Instance().GetSecret(pxutil.PortworxServiceAccountTokenSecretName, cluster.Namespace)
```

Comment:
Can we verify that the token here is different from the token we saw initially? Also, in k8s code, I see that they enforce minimum token duration of 10min. We may have to account for that if "3 min" duration is being silently overridden.

Reply:
If we set it to a smaller value (<10min), the token used by px will still be refreshed according to our configuration. It's just there will be multiple valid tokens hanging around. For example, if we refresh the token every 1min, there will be 10 valid tokens before the first one expires. If we are not enforcing user behavior, I think it's ok to set it short?